src/Misc/uvMapper.ts

```diff
@@ -0,0 +1,2299 @@
+ */
+let sortAndGetFirst = function(arr: number[][], idx: number) {
+    if (arr[idx] && arr[idx].length) {
+        arr[idx].sort((a, b) => a - b);
+
+        return arr[idx][0];
+    }
+
+    return idx;
+};
+
+/**
+ * Helper
+ */
+let intersectLineLine2d = function() {
+
+    var r, s,
+        denominator,
+        BAx, BAy, DCx, DCy;
+
+    return function(A: Vector2, B: Vector2, C: Vector2, D: Vector2) {
+                BAx = B.x - A.x;
+                BAy = B.y - A.y;
+                DCx = D.x - C.x;
+                DCy = D.y - C.y;
+                denominator = BAx * DCy - BAy * DCx;
+
+                if (Math.abs(denominator) < 1e-6) { return null; }
+                denominator = 1 / denominator;
+
+                r = ((A.y - C.y) * DCx - (A.x - C.x) * DCy) * denominator;
+                s = ((A.y - C.y) * BAx - (A.x - C.x) * BAy) * denominator;
+
+                var P = new Vector2(A.x + r * BAx, A.y + r * BAy);
+
+                if ((0 <= r) && (r <= 1) && (0 <= s) && (s <= 1)) {
+                    return P;
+                }
+
+                return null;
+            };
+}();
+
+/**
+ * Helper
+ */
+function cross(a: Vector2, b: Vector2, o: Vector2) {
+   return (a.x - o.x) * (b.y - o.y) - (a.y - o.y) * (b.x - o.x);
+}
+
+/**
+ * Helper
+ */
+function roundTo(a: number, precision: number) {
+    if (!precision) {
+        return a;
+    }
+    return Math.round(a / precision) * precision;
+}
+
+/**
+ * Helper
+ * Avoid doing:
+ *
+ * angle = atan2f(dvec[0], dvec[1]);
+ * angle_to_mat2(mat, angle);
+ *
+ * instead use a vector as a matrix.
+ */
+function mulV2V2Cw(mat: Vector2, vec: Vector2)
+{
+    return new Vector2(
+        mat.x * vec.x + mat.y * vec.y,
+        mat.y * vec.x - mat.x * vec.y
+    );
+}
+
+/**
+ * Helper
+ */
+function projectMat(vector: Vector3) {
+    let lastAxis = vector.clone().normalize();
+
+    let firstAxis: Vector3;
+
+    if (Math.abs(lastAxis.x) < SMALL_NUM && Math.abs(lastAxis.y) < SMALL_NUM) {
+        firstAxis = new Vector3(0, 1, 0);
+    } else {
+        firstAxis = new Vector3(0, 0, 1);
+    }
+
+    firstAxis = Vector3.Cross(lastAxis, firstAxis);
+    firstAxis.normalize();
+    let secondAxis = Vector3.Cross(lastAxis, firstAxis);
+    secondAxis.normalize();
+    let mat = new Matrix();
+
+    Matrix.FromXYZAxesToRef(firstAxis, secondAxis, lastAxis, mat);
+
+    return mat.transpose();
+}
+
+const USER_FILL_HOLES: number = 0;
+const USER_FILL_HOLES_QUALITY: number = 1;
+let USER_ISLAND_MARGIN: number = 0;
+let USER_STRETCH_ASPECT: boolean = false;
+const USE_PACK_BIAS: boolean = true;
+const USE_FREE_STRIP: boolean = true;
+const USE_MERGE: boolean = true;
+const SMALL_NUM: number = 1e-12;
+
+/**
+* UV Mapper for lightmaps
+* Ported from Blender by Benjamin Guignabert (https://github.com/CraigFeldspar)
+*
+* Original license can be found below :
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public License
```

Comment:
That is a problem: https://www.apache.org/licenses/GPL-compatibility.html
We cannot include GNU code inside apache2

"However, GPLv3 software cannot be included in Apache projects."